pandas/core/frame.py

```diff
@@ -360,7 +360,7 @@ class DataFrame(NDFrame, OpsMixin):
     Parameters
     ----------
     data : ndarray (structured or homogeneous), Iterable, dict, or DataFrame
-        Dict can contain Series, arrays, constants, or list-like objects. If
+        Dict can contain Series, arrays, constants, dataclass or list-like objects. If
```

Comment:
We are not adding dataclass to a dictionary, but rather as it's own object. So something like this would make more sense:
```
Iterable, dict, DataFrame or dataclass
```